cpp/modmesh/math.hpp

```diff
@@ -26,6 +26,79 @@ inline constexpr T pow(T base, std::integral_constant<size_t, N> /*unused*/)
     return pow(base, std::integral_constant<size_t, N - 1>()) * base;
 }
 
+template <typename T>
+inline constexpr T pi_v = std::enable_if_t<std::is_floating_point_v<T>, T>(3.141592653589793238462643383279502884L);
+
+template <typename T, typename = std::enable_if_t<std::is_arithmetic<T>::value>>
+struct ComplexImpl
+{
+    T real_v;
+    T imag_v;
+
+    explicit ComplexImpl(T r = 0.0, T i = 0.0)
+        : real_v(r)
+        , imag_v(i)
+    {
+    }
+    explicit ComplexImpl(const ComplexImpl & other)
+        : real_v(other.real_v)
+        , imag_v(other.imag_v)
+    {
+    }
+
+    // ComplexImpl addition
```

Comment:
Please remove the unnecessary comments.  The operator overloads are self-explaining.